test-suite/output/PrintUnivsSubgraph.out

```diff
@@ -1,4 +1,8 @@
 Set < i
     < j
-i < j
+i < j (i <(foo) j)
+
+Set < i
+    < k
+i < k (i <(foo) j <(connect) l <(baz) k)
```

Comment:
I have no clue what this means and not sure users will understand, either.  Can you add an example with some description to the doc?

Reply:
It means the constraint `i < k` is implied by the source constraints `i <j` (added by `foo`), `j < l` (added by `connect`) and `l < k` (added by `baz`)

Reply:
My initial thoughts, which may be naive:
- Lines 5-6 seem to say i < k and then line 7 repeats that with the derivation.  Are lines 5-6 necessary?
- Is the subgraph always linear (and based on transitivity)?  If so, that means the information to show is rather simple.  "subgraph" suggests more complexity.  One might call it a "Chain", though that may not be an improvement.
- It's odd to see "i <(foo) j", which is AFAICT is a mix of proper notation `i < j` with an annotation `(foo)` stuck in the middle that's only for the reader.  It may be more understandable to separate the annotations and to put each step in the chain on a separate line (akin to a proof), maybe like:
```
i < j from foo
j < l from connect
l < k from baz
```
- The symbols listed (foo, connect, baz) are always `Definition`s?  Can theorem statements be relevant?
- Does Coq ever detect impossible universe requirements?  If so, can this command be used to understand them?   (The output when you request a cyclic linking relationship for OCaml modules comes to mind as an analogy.)

... and, of course, get the explanation into the doc, not just here in the PR.

Reply:
line 5 says `Set < i` and line 6 `Set < k`

>Is the subgraph always linear (and based on transitivity)?

It is not always linear.

>It may be more understandable to separate the annotations and to put each step in the chain on a separate line (akin to a proof), maybe like:

It's already pretty verbose, you suggestion seems too verbose IMO

>The symbols listed (foo, connect, baz) are always Definitions? Can theorem statements be relevant?

Currently they can be constants or inductives or library names.

Reply:
> It's already pretty verbose, you suggestion seems too verbose IMO

Ideally the output should be understandable without having to read the manual.  It's often better to spell things out more explicitly, though not always.  There's a tradeoff between brevity and clarity.  Also, features that are understandable are more likely to be used and (I would argue) have fewer bugs and the bugs get fixed faster.

It's your choice what the output looks like, but please add some explanation of it in the doc.